### ISSUE TITLE
self-closing-void-elements as disabled by default

### ISSUE THREAD

**Issue by zorn-v** [ember-template-lint/ember-template-lint#691]
This rule has opposite opinions (e.g. https://github.com/ember-template-lint/ember-template-lint/issues/121), but by default it advices to make templates not xml compliant. I don't like it at all.
And for disable it I need to search how to disable it ? What practical purpose for this rule ?

**Comment by rwjblue** [comment]
I'm sorry you don't particularly like the default configuration. These things happen, its quite hard to please everyone. I'm glad you have figured out how to disable the rule (or change it to the opposite if you want to force valid XML).

In general, the parser we use is based on the HTML5 parsing spec so it makes sense to stick with HTML specific syntaxes. Changing the default behavior seems very unlikely, and most users are quite happy with the rule as it stands.

**Comment by rwjblue** [comment]
I'd very much welcome any tweaks/additions to the documentation to make this easier for folks that share your personal taste.

**Comment by zorn-v** [comment]
> its quite hard to please everyone.

Agree. But my opinion - standards is over opinions 😉 
You can not close some tags because browsers allow it (because lazy designers). HTML5 allow it, but "exclusion from rule" may work like "browser vendor likes".

Can you say me exact DOM for `<p><h1><div><img>` in all browsers ?

**Comment by zorn-v** [comment]
"May" should not treated as "should"

**Comment by MelSumner** [comment]
@rwjblue if we don't want to turn it off by default it would be nice if we could correct a few things. 

<img /> is valid HTML and should be allowed, shouldn't it? Or is there a different way this is determined?

**Comment by rwjblue** [comment]
> @rwjblue if we don't want to turn it off by default it would be nice if we could correct a few things.

I do not understand what you mean. Please elaborate on which things should be corrected.

> <img src="" alt="" /> is valid HTML

@MelSumner - `<img />` is corrected by browsers to `<img>`. The trailing slash in this case is meaningless, and the fact that it is corrected by the browser means that Fastboot tests (confirming that the SSR results when appended match the original SSR'ed string).

**Comment by patocallaghan** [comment]
It seems having this rule turned on by default conflicts with the Prettier hbs formatter. I opened an issue over on the Prettier repo to change it but got no response 😄 https://github.com/prettier/prettier/issues/6079

Should we just turn it off as the default in ember-template-lint instead? Whether the change happens here or over there I think the ember-template-lint default settings should try match with the Prettier hbs formatter.

**Comment by GavinJoyce** [comment]
I've created a PR to update Prettier to be consistent with the `self-closing-void-elements` rule: https://github.com/prettier/prettier/pull/6179

> Whether the change happens here or over there I think the ember-template-lint default settings should try match with the Prettier hbs formatter.

I believe that prettier for handlebars should be consistent with the `ember-template-lint` defaults.

@duailibe, a maintainer of Prettier, [doesn't seem convinced of this default](https://github.com/prettier/prettier/issues/6079#issuecomment-499114979) however. 

@duailibe, @rwjblue it would be great if we could come to a consensus here so that both `ember-template-lint` and `prettier` can work together out of the box. I'd be happy with either self closing or no self closing, with a slight preference for no self closing as that's the current default in all ember apps

**Comment by GavinJoyce** [comment]
mirroring my comment from https://github.com/prettier/prettier/issues/6079#issuecomment-499214576:

> thinking about this a little more, I think we should accept the differences between Ember, JSX and others on this rule. ember-template-lint's default rules are Ember's defaults and I think we should align prettier's rules for hbs templates to Ember's defaults. If we don't, it'll make adoption difficult as adopting prettier for hbs will mean straying from ember's conventions.

> I'm keen to work on making prettier's hbs formatting high quality and have complete support, having alignment will make that challenge much easier.

**Comment by MelSumner** [comment]
So it's valid to do both for "void elements"-  `<input type="text" />` AND `<input type="text">` should be equally okay.

However, it's not valid to do `<input type="text"></input>`. 

Here's an article: https://www.456bereastreet.com/archive/201005/void_empty_elements_and_self-closing_start_tags_in_html/

**Comment by rwjblue** [comment]
Right, I don't think we are arguing about if it is or is not valid. This part of the rule is largely stylistic (not error correcting).

**Comment by rwjblue** [comment]
I'm don't feel mega strongly on this issue, I think ensuring that no separate end tag is present, and that no children exist may be the better default position for this rule. And then we can have config to enforce self closing or not (but not have that be part of the "recommended" config).

**Comment by FabHof** [comment]
As far as I see it, disabling 'self-closing-void-elements' by default will solve all problems.
There is already a rule that handles invalid end tags of void elements.

**Comment by MelSumner** [comment]
@rwjblue some extra info: https://www.w3.org/WAI/WCAG21/Techniques/failures/F70 It looks like the closing slash should be there.

**Comment by rwjblue** [comment]
@MelSumner - I believe you are mis-reading this page. It does not say that HTML should have a trailing slash for these void elements (example 6 is **explicitly** about XHTML, not HTML).
